rpcclient/mining.go

```diff
@@ -53,9 +53,9 @@ func (r FutureGenerateResult) Receive() ([]*chainhash.Hash, error) {
 // the returned instance.
 //
 // See Generate for the blocking version and more details.
-func (c *Client) GenerateAsync(ctx context.Context, numBlocks uint32) FutureGenerateResult {
+func (c *Client) GenerateAsync(ctx context.Context, numBlocks uint32) *FutureGenerateResult {
 	cmd := chainjson.NewGenerateCmd(numBlocks)
-	return c.sendCmd(ctx, cmd)
+	return &FutureGenerateResult{ctx: ctx, c: c.sendCmd(ctx, cmd)}
```

Comment:
This works, but I think it would be safer and cleaner for `sendCmd` and `newFutureError` to return a `*cmdRes` instead of just the response channel and then all of these just reduce to a cast.

```Go
return (*FutureGetBestBlockHashResult)(c.sendCmd(ctx, cmd))
```

Then in the case an error is used, such as in `SubmitBlockAsync`, the same applies:
```Go
return (*FutureSubmitBlockResult)(newFutureError(ctx, err))
```

It's safer because it prevents the possibility of having a separate context for `sendCmd` and the promise, and, for me personally, it has the added benefit of reading nicer without the context stutter too.